torch/nn/modules/rnn.py

```diff
@@ -622,8 +635,8 @@ def reset_parameters(self):
             weight.data.uniform_(-stdv, stdv)
 
     def forward(self, input, hx):
-        self.check_batch_size(input, hx[0], '[0]')
-        self.check_batch_size(input, hx[1], '[1]')
+        self.check_forward_inputs(input, hx[0], '[0]')
+        self.check_forward_inputs(input, hx[1], '[1]')
```

Comment:
A downside of this approach is that we're doing 2 unnecessary checks here (second call repeats input checks). It would be nice to isolate input checking as a separate thing, but I guess I'm ok with leaving it as is if it doesn't increase the overhead.